packages/flutter/lib/src/widgets/drag_target.dart

```diff
@@ -821,21 +822,21 @@ class _DragAvatar<T extends Object> extends Drag {
     );
 
     // Report moves to the targets.
-    for (final _DragTargetState<T> target in _enteredTargets) {
+    for (final _DragTargetState<Object> target in _enteredTargets) {
       target.didMove(this);
     }
 
     _activeTarget = newTarget;
   }
 
-  Iterable<_DragTargetState<T>> _getDragTargets(Iterable<HitTestEntry> path) sync* {
+  Iterable<_DragTargetState<Object>> _getDragTargets(Iterable<HitTestEntry> path) sync* {
     // Look for the RenderBoxes that corresponds to the hit target (the hit target
     // widgets build RenderMetaData boxes for us for this purpose).
     for (final HitTestEntry entry in path) {
       final HitTestTarget target = entry.target;
       if (target is RenderMetaData) {
         final dynamic metaData = target.metaData;
-        if (metaData is _DragTargetState<T>)
+        if (metaData is _DragTargetState)
```

Comment:
This fixes:
`DragTarget<Object>` should accept `Draggable<int>` data

Before this change,  `metaData ` runtime is `_DragTargetState<Object>` which is not a subtype of `_DragTargetState<int>`, cause not be accepted